### ISSUE TITLE
Install Grafana with Flux

### ISSUE THREAD

**Issue by rossf7** [cncf-tags/green-reviews-tooling#16]
## Description

Install Grafana with Flux using its Helm chart
 
## Issue

Parent: https://github.com/cncf-tags/green-reviews-tooling/issues/5

**Comment by nikimanoledaki** [comment]
Thank you @rossf7, agreed! Here is a list of steps.

### Steps / Requirements
- Setup the [TF Grafana provider](https://registry.terraform.io/providers/grafana/grafana/latest/docs#creating-a-grafana-provider)
- Setup [basic auth](https://registry.terraform.io/providers/grafana/grafana/latest/docs#auth) to sign in [localhost:3000](http://localhost:3000/) using `admin:admin`
- Import the [default Kepler dashboard](Import default [dashboard](https://raw.githubusercontent.com/sustainable-computing-io/kepler-operator/v1alpha1/hack/dashboard/assets/kepler/dashboard.json) from Kepler operator repository
- [Make the dashboard public](https://grafana.com/docs/grafana/latest/dashboards/dashboard-public/#public-dashboards).

**Comment by dipankardas011** [comment]
Currect me if I am wrong
so this issue tries to solve the import of kepler grafana dashboard to the already up and running k3s cluster provided that kepler is already successfully installed

specifically these steps for automations?
https://github.com/cncf-tags/green-reviews-tooling/pull/17#issuecomment-1862363881

**Comment by dipankardas011** [comment]
Also the planned PR of installing `kube-prometheus` already installs the grafana. so may be we might need to stop the grafana as a component install?

**Comment by dipankardas011** [comment]
> Thank you @rossf7, agreed! Here is a list of suggested steps.
> 
> ### Steps / Requirements
> * Setup the [TF Grafana provider](https://registry.terraform.io/providers/grafana/grafana/latest/docs#creating-a-grafana-provider)
> * Setup [basic auth](https://registry.terraform.io/providers/grafana/grafana/latest/docs#auth) to sign in [localhost:3000](http://localhost:3000/) using `admin:admin`
> * Import the [default Kepler dashboard](https://raw.githubusercontent.com/sustainable-computing-io/kepler-operator/v1alpha1/hack/dashboard/assets/kepler/dashboard.json) from Kepler operator repository
> * [Make the dashboard public](https://grafana.com/docs/grafana/latest/dashboards/dashboard-public/#public-dashboards) using the TF Grafana provider by setting [share = "public"](https://github.com/grafana/terraform-provider-grafana/blob/master/examples/resources/grafana_dashboard_public/resource.tf#L31)

Also the planned PR of installing `kube-prometheus` already installs the grafana. so may be we might need to stop the grafana as a component install?

**Comment by AntonioDiTuri** [comment]
Yes @dipankardas011 you are right when installing prometheus we also get grafana so we could close this issue @nikimanoledaki

**Comment by dipankardas011** [comment]
so for the importing dashbaord. is it manual one time operation or we are planning for automatation of this?

**Comment by rossf7** [comment]
@dipankardas011 It needs investigation but the kube-prometheus-stack helm chart includes the grafana helm chart
https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack

The grafana helm chart can deploy a sidecar that looks for configmaps with a label.
https://github.com/grafana/helm-charts/tree/main/charts/grafana#sidecar-for-dashboards

So we could import the Kepler dashboard as a configmap managed by Flux. Later if we need our own dashboards they could also be managed by Flux. WDYT?

@nikimanoledaki You likely know this better than I do! :) Do you think this approach makes sense?

I haven't used the Grafana TF provider. Are there benefits to using it instead of the Helm chart? With Helm we will need to use Gateway API (or Ingress) to make the Grafana instance accessible from the cluster.

**Comment by dipankardas011** [comment]
now I am understanding thanks

**Comment by dipankardas011** [comment]
yes we can install the grafana dashboard without much effort

just
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: sample-grafana-dashboard
  labels:
     grafana_dashboard: "1"
data:
  kepler-dashboard.json: |-
    <raw...>
```

it works when the sidecar for the grafana in added during installing
I tested on kube-prometheus-stack
but was unable to make the kepler helm install when using `kube-prometheus-stack` up but was able to when using `kube-prometheus`

**Comment by dipankardas011** [comment]
may be kepler service monitor work for `kube-prometheus`
docs uses that 😕

**Comment by rossf7** [comment]
@dipankardas011 Nice, thanks for investigating. I think we can store it in the `monitoring` namespace since we're going to move kepler there. We can use `kepler-dashboard` for both the CM name and filename. 

It would also be good to have something in `docs/` for syncing the dashboard with upstream. I don't think it changes often so no need to automate that.

> may be kepler service monitor work for kube-prometheus docs uses that 😕

Hmm let's wait till prometheus is deployed with kube-prometheus-stack and kepler is moved to monitoring. As having everything in the monitoring ns may help with the service monitor.

cc @AntonioDiTuri

**Comment by dipankardas011** [comment]
was trying but it failed. so messaged in the kepler-project slack channel for there help
will be commenting in the slack

**Comment by dipankardas011** [comment]
other than that we can easily install the dashbaord irrespective of the prometheus stack we are using

**Comment by dipankardas011** [comment]
Also this is the grafana Dashboard which is working
